### PR TITLE
remove unnecessary interfaces from api

### PR DESCRIPTION
**Pull request by furkansenharputlu**
#245

### CODE REVIEW

**Comment by sancar** [comment]
What is the reasoning behind removing IItemEvent ?
For example, should we remove IEntryEvent interface also for same reason ?

**Comment by furkansenharputlu** [comment]
In Hazelcast Java, `ItemEvent` is an implementation in `core` package. There is no need to use interface in Go. I made same for `EntryEvent` and `MapEvent`.

**Comment by furkansenharputlu** [comment]
verify

File: internal/cluster.go [hunk 1]
```diff
@@ -351,8 +351,7 @@ func (cs *clusterService) GetMember(address core.Address) core.Member {
 	membersList := cs.members.Load().([]*protocol.Member)
 	for _, member := range membersList {
 		if *member.Address().(*protocol.Address) == *address.(*protocol.Address) {
-			copyMember := *member
-			return &copyMember
+			return member
```
Comment (SaitTalhaNisanci) [inline internal/cluster.go]: What is the reason for this change?

File: core/api.go [hunk 2]
```diff
@@ -141,30 +89,52 @@ type EntryView interface {
...
+	Name() string
+
+	// Member returns the member that fired this event.
+	Member() Member
+
+	// EventType returns the type of entry event.
+	EventType() int32
+
+	// ToString returns a string representation of this event.
+	ToString() string
```
Comment (SaitTalhaNisanci) [inline core/api.go]: Why is this `ToString` ? In Go it should be `String` which is `ToString` in Java.

File: internal/map.go [hunk 3]
```diff
@@ -539,29 +539,30 @@ func (mp *mapProxy) AddEntryListenerToKeyWithPredicate(listener interface{}, pre
...
 	valueData *serialization.Data, mergingValueData *serialization.Data, eventType int32, uuid *string,
 	numberOfAffectedEntries int32, includedValue bool, listener interface{}) {
+	member := mp.client.ClusterService.GetMemberByUUID(*uuid)
 	key, _ := mp.toObject(keyData)
 	oldValue, _ := mp.toObject(oldValueData)
 	value, _ := mp.toObject(valueData)
 	mergingValue, _ := mp.toObject(mergingValueData)
-	entryEvent := protocol.NewEntryEvent(key, oldValue, value, mergingValue, eventType, uuid)
-	mapEvent := protocol.NewMapEvent(eventType, uuid, numberOfAffectedEntries)
+	entryEvent := protocol.NewEntryEvent(mp.name, member, eventType, key, oldValue, value, mergingValue)
```
Comment (SaitTalhaNisanci) [inline internal/map.go]: Why are we changing this order?

File: core/api.go [hunk 4]
```diff
@@ -141,30 +89,52 @@ type EntryView interface {
 	TTL() time.Duration
 }
 
+// AbstractMapEvent is base for a map event.
+type AbstractMapEvent interface {
```
Comment (SaitTalhaNisanci) [inline core/api.go]: I couldnt find the usage for `AbstractMapEvent` on the client side. Can you show where it is used on the client side?

File: core/api.go [hunk 5]
```diff
@@ -260,19 +217,19 @@ type LifecycleListener interface {
...
 	// OnMessage is invoked when a message is received for the added topic. Note that topic guarantees message ordering.
 	// Therefore there is only one thread invoking OnMessage.
-	OnMessage(message TopicMessage)
+	OnMessage(message Message)
 }
 
-// TopicMessage is a message for Topic.
-type TopicMessage interface {
+// Message is a message for Topic.
+type Message interface {
```
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply core/api.go]: It is `Message` in Java.

File: internal/cluster.go (hunk 1 above)
Comment (furkansenharputlu) [reply internal/cluster.go]: Previous one seems more complex. We do not need to protect `Member`. We give it as interface to user.

File: internal/map.go (hunk 3 above)
Comment (furkansenharputlu) [reply internal/map.go]: Here, `member` and `name` were missing. I added them.

File: core/api.go (hunk 4 above)
Comment (furkansenharputlu) [reply core/api.go]: In Java server side, some interfaces are declared and used in client side. Otherwise, same interface will be created twice. Therefore, that there is no `AbstractMapEvent` on the client side is normal.

Comment (mustafaiman) [reply core/api.go]: Here `AbstractMapEvent` is an interface. On Java side, it is named abstract because it is an abstract class. 
I think it is okay to have `MapEvent` and `EntryEvent` derive from a base interface since they have common methods but naming is not right. I would change its name to `Event`.

File: internal/protocol/core.go [hunk 6]
```diff
@@ -354,13 +354,46 @@ func (e *StackTraceElement) LineNumber() int32 {
 	return e.lineNumber
 }
 
+type AbstractMapEvent struct {
+	name      *string
```
Comment (SaitTalhaNisanci) [inline internal/protocol/core.go]: We should use `string` instead of `*string`here and in the rest of this PR.

File: core/api.go [hunk 7]
```diff
@@ -220,26 +177,26 @@ type EntryMergedListener interface {
 	EntryMerged(EntryEvent)
```
Comment (SaitTalhaNisanci) [inline core/api.go]: event
